src/main.rs

```diff
@@ -1,3 +1,26 @@
-fn main() {
-    println!("ello");
+
+use gtree;
+use std::fs;
+use std::fs::File;
+use gitignore;
+use std::path::Path;
+
+fn main() -> std::io::Result<()> {
+    let maybe_file = File::open(".gitignore");
+    match maybe_file {
+        Ok(_) => {
+            let gitignore_path = Path::new(".gitignore");
+            let gitignore_from_lib = gitignore::File::new(gitignore_path).unwrap();
```

Comment:
This is good for a prototype, but in the general case `.gitignore` files are allowed at any level of the repository so you need to handle looking for them and adding them to the list of checks for ignored files and then popping them off the stack as you recurse back out.